src/pyscaffold/shell.py

```diff
@@ -159,22 +175,29 @@ def get_executable(
     candidates = list(Path(prefix).resolve().glob(f"*/{name}*"))
     # ^  this works in virtual envs and both Windows and POSIX
     if candidates:
-        path = [str(f.parent) for f in sorted(candidates, key=lambda p: len(str(p)))]
-        return shutil.which(name, path=os.pathsep.join(path))
+        path = (f.parent for f in sorted(candidates, key=lambda p: len(str(p))))
+        return shutil.which(name, path=os.pathsep.join(str(p) for p in path))
         # ^  which will guarantee we find an executable and not only a regular file
 
     return None
 
 
 def get_command(
-    name: str, prefix: PathLike = sys.prefix, include_path=True, **kwargs
+    name: str, prefix: PathLike = sys.prefix, include_path=True, shell=True, **kwargs
 ) -> Optional[ShellCommand]:
     """Similar to :obj:`get_executable` but return an instance of :obj:`ShellCommand`
     if it is there to be found.
     Additional kwargs will be passed to the :obj:`ShellCommand` constructor.
     """
     executable = get_executable(name, prefix, include_path)
-    return ShellCommand(executable, **kwargs) if executable else None
+    if not executable:
+        return None
+
+    if shell:
+        executable = _quote(executable)
```

Comment:
Quoting here should be safe, since we are automatically obtaining the executable path, and checking for `shell`